messari-cli/src/commands/init.rs

```diff
@@ -196,6 +201,58 @@ fn create_substreams_project(
     let mut project_makefile_contents = MakeFile::new(&project_makefile);
     project_makefile_contents.add_build_operation();
 
+    let mut yaml_contents = SubstreamsYaml::new(project_name.as_str(), &substreams_yaml);
```

Comment:
I provided this feedback in the previous PR but wanted to ask again here: what's the reasoning on using a procedural approach instead of a declarative approach here? It seems like this is growing in complexity which will make it more and more difficult to maintain. Also, we now have a number of static files that are copied as templates, it would make sense for all files to be generated from templates for consistency.

Reply:
As discussed today and yesterday in person the idea here is that we want procedural generation to allow us to also use the code for mutating existing documents when adding our extra data. This will be important later on especially for when we start to do any extra code generation around the modification of map or store methods, including also the changes based around protos which also need to be reflected in the yaml document. The reason for the inconsistency with the template files is that, in particular, the two files erc20.json and lib.rs won't be templates for long. Soon it would be good to create them procedurally based on user requirements to give a greater setup leading to less work for the developer.